// @progress/kendo-angular-utils/esm2020/drag-and-drop/droptarget.directive.mjs
/**-----------------------------------------------------------------------------------------
* Copyright © 2023 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import { Directive, ElementRef, EventEmitter, NgZone, Output } from "@angular/core";
import { validatePackage } from '@progress/kendo-licensing';
import { packageMetadata } from '../package-metadata';
import { DragStateService } from "./drag-state.service";
import { DropTargetEvent } from './events/drop-target-event';
import { isPresent } from "./util";
import * as i0 from "@angular/core";
import * as i1 from "./drag-state.service";
/**
 * Represents the Kendo UI DropTarget directive for Angular.
 */
export class DropTargetDirective {
    constructor(service, element, ngZone) {
        this.service = service;
        this.element = element;
        this.ngZone = ngZone;
        /**
         * Fires when a DragTarget element enters the DropTarget.
         */
        this.dragEnter = new EventEmitter();
        /**
         * Fires when a DragTarget element is being dragged over the DropTarget.
         */
        this.dragOver = new EventEmitter();
        /**
         * Fires when a DragTarget element leaves the DropTarget.
         */
        this.dragLeave = new EventEmitter();
        /**
         * Fires when a DragTarget element is dropped over the DropTarget.
         */
        this.drop = new EventEmitter();
        validatePackage(packageMetadata);
    }
    ngOnInit() {
        this.initializeDropTarget();
        this.service.dropTargets.push(this.dropTarget);
    }
    /**
     * @hidden
     */
    handleDragEnter(event) {
        if (!this.service.dragTarget) {
            return;
        }
        this.service.dropTarget = this.dropTarget;
        this.ngZone.run(() => {
            this.emitEvent('dragEnter', event);
        });
    }
    /**
     * @hidden
     */
    handleDragLeave(event) {
        this.service.dropTarget = null;
        if (!this.service.dragTarget) {
            return;
        }
        this.ngZone.run(() => {
            this.emitEvent('dragLeave', event);
        });
    }
    /**
     * @hidden
     */
    handleDragOver(event) {
        if (!this.service.dragTarget) {
            return;
        }
        this.ngZone.run(() => {
            this.emitEvent('dragOver', event);
        });
    }
    /**
     * @hidden
     */
    handleDrop(event) {
        this.ngZone.run(() => {
            this.emitEvent('drop', event);
        });
    }
    initializeDropTarget() {
        this.dropTarget = {
            element: this.element.nativeElement,
            onDragEnter: this.handleDragEnter.bind(this),
            onDragLeave: this.handleDragLeave.bind(this),
            onDragOver: this.handleDragOver.bind(this),
            onDrop: this.handleDrop.bind(this)
        };
    }
    emitEvent(event, normalizedEvent) {
        const eventProps = {
            dropTarget: this.element.nativeElement,
            dragTarget: this.service.dragTarget?.element,
            dragEvent: normalizedEvent,
            dragData: this.service.dragData
        };
        if (isPresent(this.service.dragTarget?.hint)) {
            eventProps.hintElement = this.service.dragTarget.element;
        }
        const eventArgs = new DropTargetEvent(eventProps);
        this[event].emit(eventArgs);
    }
}
DropTargetDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: DropTargetDirective, deps: [{ token: i1.DragStateService }, { token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive });
DropTargetDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.3.12", type: DropTargetDirective, selector: "[kendoDropTarget]", outputs: { dragEnter: "dragEnter", dragOver: "dragOver", dragLeave: "dragLeave", drop: "drop" }, exportAs: ["kendoDropTarget"], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: DropTargetDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[kendoDropTarget]',
                    exportAs: 'kendoDropTarget'
                }]
        }], ctorParameters: function () { return [{ type: i1.DragStateService }, { type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { dragEnter: [{
                type: Output
            }], dragOver: [{
                type: Output
            }], dragLeave: [{
                type: Output
            }], drop: [{
                type: Output
            }] } });
