// @progress/kendo-angular-grid/esm2020/pager/pager-prev-buttons.component.mjs
/**-----------------------------------------------------------------------------------------
* Copyright © 2023 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import { NavigationService } from './../navigation/navigation.service';
import { Component, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
import { PagerContextService } from "./pager-context.service";
import { PagerElementComponent } from './pager-element.component';
import { LocalizationService } from '@progress/kendo-angular-l10n';
import { caretAltLeftIcon, caretAltToLeftIcon } from '@progress/kendo-svg-icons';
import * as i0 from "@angular/core";
import * as i1 from "@progress/kendo-angular-l10n";
import * as i2 from "./pager-context.service";
import * as i3 from "./../navigation/navigation.service";
import * as i4 from "@progress/kendo-angular-buttons";
import * as i5 from "../navigation/focusable.directive";
/**
 * Displays buttons for navigating to the first and to the previous page ([see example](slug:pager_template_grid#toc-using-built-in-pager-components)).
 */
export class PagerPrevButtonsComponent extends PagerElementComponent {
    constructor(localization, pagerContext, cd, navigationService) {
        super(localization, pagerContext, cd);
        this.navigationService = navigationService;
        this.caretAltLeftIcon = caretAltLeftIcon;
        this.caretAltToLeftIcon = caretAltToLeftIcon;
    }
    /**
     * @hidden
     *
     * @readonly
     * @type {boolean}
     * @memberOf PagerPrevButtonsComponent
     */
    get disabled() {
        return this.currentPage === 1 || !this.total;
    }
    onChanges({ total, skip, pageSize }) {
        this.total = total;
        this.skip = skip;
        this.pageSize = pageSize;
        this.cd.markForCheck();
    }
}
PagerPrevButtonsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: PagerPrevButtonsComponent, deps: [{ token: i1.LocalizationService }, { token: i2.PagerContextService }, { token: i0.ChangeDetectorRef }, { token: i3.NavigationService }], target: i0.ɵɵFactoryTarget.Component });
PagerPrevButtonsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.3.12", type: PagerPrevButtonsComponent, selector: "kendo-pager-prev-buttons", usesInheritance: true, ngImport: i0, template: `
        <button
            [kendoGridFocusable]="!disabled"
            kendoButton
            type="button"
            [title]="textFor('pagerFirstPage')"
            [attr.aria-label]="textFor('pagerFirstPage')"
            (click)="currentPage !== 1 ? changePage(0) : false"
            role="button"
            icon="caret-alt-to-left"
            [svgIcon]="caretAltToLeftIcon"
            class="k-pager-nav k-pager-first"
            [disabled]="disabled"
            fillMode="flat"
            rounded="none"
            [size]="size">
        </button>
        <button
            kendoButton
            [kendoGridFocusable]="!disabled"
            type="button"
            class="k-pager-nav"
            [disabled]="disabled"
            icon="caret-alt-left"
            [svgIcon]="caretAltLeftIcon"
            fillMode="flat"
            rounded="none"
            [size]="size"
            [title]="textFor('pagerPreviousPage')"
            [attr.aria-label]="textFor('pagerPreviousPage')"
            (click)="currentPage !== 1 ? changePage(currentPage-2) : false">
        </button>
    `, isInline: true, components: [{ type: i4.Button, selector: "button[kendoButton], span[kendoButton], kendo-button", inputs: ["toggleable", "togglable", "selected", "tabIndex", "imageUrl", "iconClass", "icon", "disabled", "size", "rounded", "fillMode", "themeColor", "svgIcon", "role", "primary", "look"], outputs: ["selectedChange", "click"], exportAs: ["kendoButton"] }], directives: [{ type: i5.FocusableDirective, selector: "[kendoGridFocusable],\n        [kendoGridEditCommand],\n        [kendoGridRemoveCommand],\n        [kendoGridSaveCommand],\n        [kendoGridCancelCommand],\n        [kendoGridSelectionCheckbox]\n    ", inputs: ["kendoGridFocusable"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: PagerPrevButtonsComponent, decorators: [{
            type: Component,
            args: [{
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    selector: 'kendo-pager-prev-buttons',
                    template: `
        <button
            [kendoGridFocusable]="!disabled"
            kendoButton
            type="button"
            [title]="textFor('pagerFirstPage')"
            [attr.aria-label]="textFor('pagerFirstPage')"
            (click)="currentPage !== 1 ? changePage(0) : false"
            role="button"
            icon="caret-alt-to-left"
            [svgIcon]="caretAltToLeftIcon"
            class="k-pager-nav k-pager-first"
            [disabled]="disabled"
            fillMode="flat"
            rounded="none"
            [size]="size">
        </button>
        <button
            kendoButton
            [kendoGridFocusable]="!disabled"
            type="button"
            class="k-pager-nav"
            [disabled]="disabled"
            icon="caret-alt-left"
            [svgIcon]="caretAltLeftIcon"
            fillMode="flat"
            rounded="none"
            [size]="size"
            [title]="textFor('pagerPreviousPage')"
            [attr.aria-label]="textFor('pagerPreviousPage')"
            (click)="currentPage !== 1 ? changePage(currentPage-2) : false">
        </button>
    `
                }]
        }], ctorParameters: function () { return [{ type: i1.LocalizationService }, { type: i2.PagerContextService }, { type: i0.ChangeDetectorRef }, { type: i3.NavigationService }]; } });
